daemon/api_themes.go

```diff
@@ -0,0 +1,245 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/store"
+	"github.com/snapcore/snapd/strutil"
+)
+
+var (
+	themesCmd = &Command{
+		Path:   "/v2/themes",
+		UserOK: true,
+		GET:    checkThemes,
+		POST:   installThemes,
+	}
+)
+
+type themeStatus string
+
+const (
+	themeInstalled   themeStatus = "installed"
+	themeAvailable   themeStatus = "available"
+	themeUnavailable themeStatus = "unavailable"
+)
+
+type themeStatusResponse struct {
+	GtkThemes   map[string]themeStatus `json:"gtk-themes"`
```

Comment:
Do we need to differentiate gtk2 from gtk3 here?

Reply:
No.  Both GTK 2 and GTK 3 use the same configuration value to set their theme.  So we can't easily say "this theme name is only needed for GTK 2 apps", for instance.

Where there are matched GTK 2 and GTK 3 themes, they are generally produced by the same designers and distributed together.  So it's not clear that splitting them up into separate snaps would be particularly helpful here.